lib/evmone/eof.hpp

```diff
@@ -5,17 +5,29 @@
 
 #include <evmc/evmc.h>
 #include <evmc/utils.h>
+#include <array>
 #include <cassert>
 #include <cstddef>
 #include <cstdint>
 #include <string>
+#include <variant>
 #include <vector>
 
 namespace evmone
 {
+constexpr uint8_t MAGIC[] = {0xef, 0x00};
```

Comment:
I didn't realize it will drag along all the constants into the header... What if instead of making `validation_section_headers()` public as is, we make a public wrapper of it that returns only full container size that we need? maybe like `std::variant<size_t, EOFValidationError> validate_eof_section_headers(bytes_view)`

Reply:
Also `MAGIC` doesn't have to be in the header in any case, does it?

Reply:
re `MAGIC` - yup, I grabbed too much. Re the fist comment, see my other comment and the idea to use `validate_header` instead.